huaweicloud/services/coc/resource_huaweicloud_coc_script_order_operation.go

```diff
@@ -0,0 +1,103 @@
+package coc
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API COC PUT /v1/job/script/orders/{execute_uuid}/operation
+func ResourceScriptOrderOperation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScriptOrderOperationCreate,
+		ReadContext:   resourceScriptOrderOperationRead,
+		UpdateContext: resourceScriptOrderOperationUpdate,
+		DeleteContext: resourceScriptOrderOperationDelete,
+
+		Schema: map[string]*schema.Schema{
+			"execute_uuid": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"batch_index": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"operation_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func buildScriptOrderOperationCreateOpts(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"batch_index":    utils.ValueIgnoreEmpty(d.Get("batch_index")),
+		"instance_id":    utils.ValueIgnoreEmpty(d.Get("instance_id")),
+		"operation_type": utils.ValueIgnoreEmpty(d.Get("operation_type")),
```

Comment:
no need to ignoreValueEmpty for required param. 
```suggestion
		"operation_type": d.Get("operation_type"),
```